gapis/api/vulkan/profiling_layers.go

```diff
@@ -0,0 +1,94 @@
+// Copyright (C) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vulkan
+
+import (
+	"context"
+
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/api/transform"
+)
+
+var renderStagesLayer = "VkRenderStagesProducer"
+
+type profilingLayers struct{}
+
+func (t *profilingLayers) Transform(ctx context.Context, id api.CmdID, cmd api.Cmd, out transform.Writer) {
+	ctx = log.Enter(ctx, "ProfilingLayers")
+
+	s := out.State()
+	cb := CommandBuilder{Thread: cmd.Thread(), Arena: out.State().Arena}
+	l := s.MemoryLayout
+	allocated := []api.AllocResult{}
+	defer func() {
+		for _, d := range allocated {
+			d.Free()
+		}
+	}()
+	mustAlloc := func(ctx context.Context, v ...interface{}) api.AllocResult {
+		res := s.AllocDataOrPanic(ctx, v...)
+		allocated = append(allocated, res)
+		return res
+	}
+
+	switch cmd := cmd.(type) {
+	case *VkCreateInstance:
```

Comment:
This will cause the replay to fail if the layer is not available. I would prefer if we either don't do anything if the layer is not available. At least add a TODO for now.